misk/src/main/kotlin/misk/web/actions/NotFoundAction.kt

```diff
@@ -38,4 +50,40 @@ class NotFoundAction @Inject constructor() : WebAction {
       )
     }
   }
+
+  /** Generates a message for a client that hints what may have been wrong with their request. */
+  private fun notFoundMessage(path: String, httpCall: HttpCall) = buildString {
+    val boundActions = servletProvider.get().boundActions
+    val alternativeActions = boundActions.mapNotNull {
+      it.matchByUrl(httpCall.url)
+    }.filterNot { it.action.pathPattern.matchesWildcardPath }
```

Comment:
does this prevent it from matching useful things?

Reply:
It might. 

I did this for two reasons:
1. Prevent from matching on `/{path: .*}` routes, which aren't useful suggestions.
2. Concerns about leaking too much information.

As @adrw mentioned:
> Curious about the security implications around giving more information on endpoints through the error messages. Otherwise looks good 👍

I don't want to suggest too much &mdash; just things that look like the caller is definitely aware of. If a client is making a call to weird wildcard path, they might not be provided any suggestions, but they _should_ know about what the endpoint they are trying to call requires... but they might not realize they forgot to set the right content-type, so context is still helpful.

Reply:
Would there be a way to also only recommend paths that they have sufficient authentication to see?

Reply:
I'd expect so. Exploring that as an option.

Reply:
The information leakage is very high. Not going to delve into this. Any info that would be leaked by this is already discoverable. 

Someone who really cares about this could deploy an interceptor or an endpoint that supercedes the behaviour of the NotFoundAction.